common/functions/src/expressions/mod.rs

```diff
@@ -6,5 +6,9 @@
 mod cast_test;
 
 mod cast;
+mod expressions;
+mod substring;
```

Comment:
Is it better to put function substring to directory udfs?

Reply:
fixed

Reply:
@tlightsky 
Thanks for the work.
We may have many string functions in future, it would be better to create a directory 'strings' for them, looks more clearly.
'udfs' for some uncategorized functions.
Cc @sundy-li what do you think, sundy?

Reply:
It's good to have string subdirectory. The main idea is to keep ExpressionPlan as thin as it can,  so we can avoid lots of  match expressions branches.

Reply:
Hi @BohuTANG @sundy-li ,
moved `substring` to `strings`